File: Week 01/id_503/leetcode_26_503.js
/**
 * 统计重复数量法
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function (nums) {
	let count = 0;
	for (let i = 0; i < nums.length; i++) {
		if (nums[i] !== nums[i - 1]) {
			nums[i - count] = nums[i];
			continue;
		}

		count++;
	}

	return nums.length - count;
};

File: Week 01/id_503/leetcode_26_503.go
package week01

// 双指针法
func removeDuplicates(nums []int) int {
	snow, fast := 0, 1
	for ; fast < len(nums); fast++ {
		if nums[fast] != nums[fast-1] {
			snow++
			nums[snow] = nums[fast]
		}
	}

	return snow + 1
}
